keps/sig-node/3815-pod-condition-for-configuration-errors/README.md

```diff
@@ -0,0 +1,1010 @@
+A new condition will provide a quick way to verify if a pod has detected an unrecoverable configuration error.
+
+## Motivation
+
+Users may not have deep knowledge of Kubernetes.  It is common occurrence to find users submitting Pods with configuration errors (invalid images, wrong image names, missing volumes, missing secrets, wrong volume/secret for config map).  These cases cause the Pod to be stuck in pending and there is code required to remove these pods or user input to remove.
+
+It would be ideal to have a standard condition for these common configuration errors.  These errors can be fixed via human interaction but they can sometimes be difficult to programatically fix.  A goal of having a standard condition for these configuration errors would be for third-party controllers and/or Kubernetes controllers to start using a single way of detecting fixable configuration errors.  For example, in the Armada project, we utilize container statuses and events to deduce these type of configuration errors.  Events are best-effort so they can easily be missed.
+
+It is common to have third-party controllers that monitor conditions to verify the stability of your workloads.  Currently, to deduce configuration errors, you have to either monitor events, view container statuses, or sometimes view conditions based on when your pod got stuck.  A stable condition for configuration errors will enhance the experience for third-party developers as they can programatically determine a failure with a stable condition.  
+
+### Goals
+
+<<[UNRESOLVED @kannon92 ]>>
+Condition Name is up to debate:
+
+a) PodFailed
+b) PodFailedToStart
+c) PodHasConfigurationIssue
+
+Reviewers, please put some other options here.
+<<[/UNRESOLVED]>>
+
+- Add a new condition FailingToStart to detect failures in configuration.
+
+For future cases that we want to add to to this condition, the suggestion would be to consider cases where pods are forever stuck in the pending state.  We want to avoid transient pods that will eventually resolve themselves.  The general rule could be that pods that require manually intervention to fix could be added as cases for this condition.  
+
+### Non-Goals
+
+- Configuration errors that happen during Sandbox creation (Mounting volumes, Network creation) will not be included as part of this condition
+  - PodReadyToStartContainers condition actually covers these cases so there is no need to cover them in this KEP.
+- Configuration errors that happen during Init containers.
+  - These errors are already captured by the Initialized condition as when an init container stage is passed, the Intialized condition would be set to true.
+
+## Proposal
+
+- Add a PodFailedToStart Condition for the following cases:
+
+  - Invalid Image Name
+  - Image not found locally when ImagePullPolicy is set to never.
+  - Existing config map but invalid key reference
+  - Container lifecycle hook failures
```

Comment:
Wondering if we are going to cover the following cases:

- Incorrect node selector
- Untolerated taints
- Requested resource too high for the selected node (cpu, memory)

Reply:
Those are already covered by the Scheduled=False condition.